src/test/java/org/takes/http/BkBasicTest.java

```diff
@@ -77,4 +86,26 @@ public void handlesSocket() throws IOException {
             Matchers.containsString("Hello world")
         );
     }
+
+    /**
+     * BkBasic can return HTTP status 404 when accessing invalid URL.
+     * @throws IOException if any I/O error occurs.
+     */
+    @Test
+    public void returns404OnInvalidUrl() throws IOException {
```

Comment:
@cyberone Let's declare `throws Exception` as per our [code standards](http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#test-method-names).